Classes/Queue/DoctrineQueue.php

```diff
@@ -191,15 +200,38 @@ protected function reserveMessage(?int $timeout = null): ?Message
         }
         $this->reconnectDatabaseConnection();
 
+        $qb = $this->connection->createQueryBuilder();
+        $qb
+            ->select('*')
+            ->from($this->connection->quoteIdentifier($this->tableName))
+            ->where(
+                $qb->expr()->and(
+                    $qb->expr()->eq('state', $qb->expr()->literal('ready')),
+                    $this->getScheduledQueryConstraint($qb)
+                )
+            )
+            ->orderBy('id')
+            ->setMaxResults(1);
         $startTime = time();
         do {
             try {
-                $row = $this->connection->fetchAssociative("SELECT * FROM {$this->connection->quoteIdentifier($this->tableName)} WHERE state = 'ready' AND {$this->getScheduledQueryConstraint()} ORDER BY id ASC LIMIT 1");
+                $row = $qb->execute()->fetchAssociative();
             } catch (TableNotFoundException $exception) {
                 throw new \RuntimeException(sprintf('The queue table "%s" could not be found. Did you run ./flow queue:setup "%s"?', $this->tableName, $this->name), 1469117906, $exception);
             }
             if ($row !== false) {
-                $numberOfUpdatedRows = (int)$this->connection->executeStatement("UPDATE {$this->connection->quoteIdentifier($this->tableName)} SET state = 'reserved' WHERE id = :id AND state = 'ready' AND {$this->getScheduledQueryConstraint()}", ['id' => (int)$row['id']]);
+                $innerQueryBuilder = $this->connection->createQueryBuilder();
```

Comment:
why is that one "inner"?

Reply:
It's in a loop and I need the "original" `$qb` to stay intact.

Reply:
I'lll move out out of the loop…